ex6.sh

```diff
@@ -0,0 +1,11 @@
+NUMBER=${1:-50}
+FOLDER_NAME="zero_$(date +%Y%m%d)_files"
+mkdir "$FOLDER_NAME"
+for ((i = 1; i <= NUMBER; i++)); do
+  FILE_NAME="File_${i}.dat"
+  FILE_CONTENT=$(printf "%0${i}d" 0)
+  echo "$FILE_CONTENT" >"$FOLDER_NAME/$FILE_NAME"

```

Comment:
Notice that the demand was to save nullbytes (\x00) in the file and not ascii char 0 (\x30)